app/controllers/vm_common.rb

```diff
@@ -1370,7 +1370,7 @@ def console_before_task(console_type)
   # Task complete, show error or launch console using VNC/MKS/VMRC task info
   def console_after_task(console_type)
     miq_task = MiqTask.find(params[:task_id])
-    if miq_task.status == "Error" || miq_task.task_results.blank?
+    unless miq_task.results_ready?
```

Comment:
> Is there a non-Error status they can be in besides Ok? I don't remember if there's a running...

There are two levels of status. There is Taks#state (initialized, queued, active, finished) and then there is Task#status (ok, warn, error, timeout, expired).

Fortunately, we don't need to care about this question. If the task is still running, we can expect it to have `task_result.blank? == true`.

The beauty of this change is that it makes things consistent and more strict while not breaking the behavior (unless there is a buggy worker that will get exposed).

> Can you add tests for the results_ready? method based on how all of these controller conditions currently work?

I was thinking about this when creating the pr, but then I discarded the idea. I felt like these tests would only test single expression `status == STATUS_OK && !task_results.blank?` and nothing else. What kind of bugs would such test reveal?